sccameraview/src/main/java/com/example/sccameraview/BaseCameraView.java

```diff
@@ -53,20 +61,32 @@ public void setCalculatedHeight(int calculatedHeight) {
     }
 
     CamcorderProfile getCamcorderProfile() {
+        if (CamcorderProfile.hasProfile(cameraId, CamcorderProfile.QUALITY_HIGH)) {
+            return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_HIGH);
+        }
         if (CamcorderProfile.hasProfile(cameraId, CamcorderProfile.QUALITY_480P)) {
             return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_480P);
         }
-
         return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_LOW);
     }
 
     abstract void openCamera();
 
-    private class SurfaceTextureListener implements TextureView.SurfaceTextureListener {
+    public abstract void switchCamera();
+
+    private class SurfaceTextureListener implements TextureView.SurfaceTextureListener, ActivityCompat.OnRequestPermissionsResultCallback {
 
         @Override
         public void onSurfaceTextureAvailable(SurfaceTexture surface, int width, int height) {
-            openCamera();
+            if (!(ContextCompat.checkSelfPermission(getContext(), Manifest.permission.WRITE_EXTERNAL_STORAGE) == PackageManager.PERMISSION_GRANTED &&
```

Comment:
We do not need permission check in this library, this is the responsibility of the client.